WooCommerce/Classes/ViewRelated/Dashboard/Reviews/ReviewsDashboardCardViewModel.swift

```diff
@@ -19,30 +19,185 @@ final class ReviewsDashboardCardViewModel: ObservableObject {
         case approved
     }
 
+    @Published private(set) var data: [ReviewViewModel] = []
+    private var reviewProducts: [Product] {
+        return productsResultsController.fetchedObjects
+    }
     @Published private(set) var syncingData = false
     @Published private(set) var syncingError: Error?
 
-    private let siteID: Int64
     private let stores: StoresManager
-    private let storage: StorageManagerType
+    private let storageManager: StorageManagerType
     private let analytics: Analytics
 
+    public let siteID: Int64
     public let filters: [ReviewsFilter] = [.all, .hold, .approved]
 
+    private var productsResultsController: ResultsController<StorageProduct>
+
     init(siteID: Int64,
          stores: StoresManager = ServiceLocator.stores,
-         storage: StorageManagerType = ServiceLocator.storageManager,
+         storageManager: StorageManagerType = ServiceLocator.storageManager,
          analytics: Analytics = ServiceLocator.analytics) {
         self.siteID = siteID
         self.stores = stores
-        self.storage = storage
+        self.storageManager = storageManager
         self.analytics = analytics
+
+        self.productsResultsController = ResultsController<StorageProduct>(storageManager: storageManager,
+                                                                           matching: nil,
+                                                                           sortedBy: [])
```

Comment:
nit: We could mention the `fetchLimit: Constants.numberOfItems` here as well.

Reply:
Good thinking. I had assumed that when the predicate is updated and it's requesting up to 3 product IDs, it would always return up to 3 items anyway since they're distinct. But I agree it's safer to limit it here too. Added in https://github.com/woocommerce/woocommerce-ios/pull/12752/commits/37fc4823dc944ec8a9b41c3a3cf4e11032b97085